kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -125,7 +127,7 @@ public KafkaProxy startup() {
                     .map(c -> new HostPort(c.getEffectiveBindAddress(), c.getEffectivePort()));
             portConflictDefector.validate(virtualClusterModels, managementHostPort);
 
-            var availableCores = Runtime.getRuntime().availableProcessors();
+            int availableCores = resolveThreadCount().orElse(Runtime.getRuntime().availableProcessors());
```

Comment:
this currently feeds both management and proxy, I think it should split into something like:

```
int proxyWorkerThreadCount = resolveThreadCount(NetworkDefinition::proxy).orElse(Runtime.getRuntime().availableProcessors());
int managementWorkerThreadCount = resolveThreadCount(NetworkDefinition::management).orElse(Runtime.getRuntime().availableProcessors());
```